src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -258,11 +256,6 @@ public abstract class GenerateProtoTask extends DefaultTask {
     this.outputBaseDir = outputBaseDir
   }
 
-  @OutputDirectory
```

Comment:
What problem are we trying to solve with this logic? Looks too complicated.

Reply:
All the outputs need to be tracked. If you remove the output base dir as an output, then you must track all the individual files/directories. But `getOutputSourceDirectories()` doesn't return any of the files. Thus you need some other mechanism to expose the output files. There's multiple ways to do that.

Reply:
I see comments in the code
> protoc is capable of output generated files directly to a JAR file or ZIP archive if the output location ends with .jar/.zip

However, I do not see any documentation of this feature in the protobuf-gradle-plugin. I also do not see any tests for this feature.

Reply:
> Not entirely. DescriptorSetOptions.path is an @OutputFile, but it is only used if the user overrides the path. If the user uses the default path then that path would be untracked.

Sounds like a bug, doesn't it?

Reply:
> I do not see any documentation of this feature in the protobuf-gradle-plugin

All the background I see: https://github.com/google/protobuf-gradle-plugin/pull/480 and https://github.com/google/protobuf-gradle-plugin/pull/533 .
> Sounds like a bug, doesn't it?

Not on master. If there is an output descriptor, then it is either a child of the base directory or it is the user-configured name. This PR removes the base directory as an output so that introduces a bug.